mm2src/lp_swap.rs

```diff
@@ -264,10 +269,52 @@ async fn recv_swap_msg<T>(
 /// in order to give different and/or heavy communication channels a chance.
 const BASIC_COMM_TIMEOUT: u64 = 90;
 
+#[cfg(not(feature = "custom-swap-locktime"))]
 /// Default atomic swap payment locktime, in seconds.
 /// Maker sends payment with LOCKTIME * 2
 /// Taker sends payment with LOCKTIME
-pub const PAYMENT_LOCKTIME: u64 = 3600 * 2 + 300 * 2;
+const PAYMENT_LOCKTIME: u64 = 3600 * 2 + 300 * 2;
+
+#[cfg(feature = "custom-swap-locktime")]
+lazy_static! {
+    /// Default atomic swap payment locktime, in seconds.
+    /// Maker sends payment with LOCKTIME * 2
+    /// Taker sends payment with LOCKTIME
+    static ref PAYMENT_LOCKTIME: u64 = get_payment_locktime_from_mm2config();
+}
+
+#[cfg(feature = "custom-swap-locktime")]
+/// Reads `payment_locktime` from cli or MM2.json configuration and returns it.
+/// Returns 900 when `payment_locktime` is invalid or not provided.
+fn get_payment_locktime_from_mm2config() -> u64 {
+    let args: Vec<OsString> = env::args_os().collect();
+    let first_arg = args.get(1).and_then(|arg| arg.to_str());
+    let conf = super::get_mm2config(first_arg).unwrap();
```

Comment:
Maybe we should return the default value instead of panicking if unwrap fails. I know it shouldn't fail but it's always better to avoid `unwrap()` whenever we can.

Reply:
Also, a question for @sergeyboyko0791, do you think getting the `conf` from `ctx` using `from_ffi_handle` https://github.com/KomodoPlatform/atomicDEX-API/blob/c769f38339600e45ca63b61193e0009f0e2e306b/mm2src/common/mm_ctx.rs#L444 can be better than getting it from `args`. I am asking this question because I am unsure which is the better approach 🙂

Reply:
It looks like a better option than reading the config file. But there is a problem related to the fact that there is a static [CTX](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/mm2_lib.rs#L33) variable in `mm2_lib` crate only.
We can add the same variable somewhere in `mm2` crate to keep the FFI handle of the current instance, but it should be initialized once on the MarketMaker starts.
In the same time we can just initialize `PAYMENT_LOCKTIME` once on MarketMaker starts. What do you think?

Reply:
> In the same time we can just initialize PAYMENT_LOCKTIME once on MarketMaker starts.

This is the best solution of course, `PAYMENT_LOCKTIME` when `custom-swap-locktime` is enabled https://github.com/KomodoPlatform/atomicDEX-API/blob/c769f38339600e45ca63b61193e0009f0e2e306b/mm2src/lp_swap.rs#L278-L284 will have to be imported to `mm2.rs` but it's not an issue I guess.